gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -697,19 +704,51 @@ _message_handler (void *thread_data)
   }
 
   if (cmd_data.cmd == _TENSOR_QUERY_CMD_REQUEST_INFO) {
-    GstTensorsConfig *config = &cmd_data.data_info.config;
-    if ((gst_tensors_config_is_flexible (config) &&
-            gst_tensors_config_is_flexible (&_sdata->src_config)) ||
-        gst_tensors_info_is_equal (&config->info, &_sdata->src_config.info)) {
+    GstCaps *sdata_caps, *cmd_caps;
+    GstStructure *structure;
+    gboolean result = FALSE;
+
+    sdata_caps = gst_caps_from_string (_sdata->src_caps_str);
+    cmd_caps = gst_caps_from_string ((char *) cmd_data.data.data);
+    /** Server frame rate may vary. Let's skip comparing the framerate. */
+    gst_caps_set_simple (cmd_caps, "framerate", GST_TYPE_FRACTION, 0, 1, NULL);
+    gst_caps_set_simple (sdata_caps, "framerate", GST_TYPE_FRACTION, 0, 1,
+        NULL);
+
+    structure = gst_caps_get_structure (sdata_caps, 0);
+
+    free (cmd_data.data.data);
+
+    if (gst_structure_is_tensor_stream (structure)) {
+      GstTensorsConfig sdata_config, cmd_config;
+      GstStructure *cmd_structure;
+
+      gst_tensors_config_init (&cmd_config);
+      gst_tensors_config_init (&sdata_config);
+      cmd_structure = gst_caps_get_structure (cmd_caps, 0);
+
+      gst_tensors_config_from_structure (&sdata_config, structure);
+      gst_tensors_config_from_structure (&cmd_config, cmd_structure);
+
+      result = gst_tensors_config_is_equal (&sdata_config, &cmd_config);
+    }
+
+    if (result || gst_caps_can_intersect (cmd_caps, sdata_caps)) {
       cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
-      /* respond sink config */
-      gst_tensors_config_copy (config, &_sdata->sink_config);
+      cmd_data.data.data = (uint8_t *) _sdata->sink_caps_str;
     } else {
-      /* respond deny with src config */
-      nns_logw ("tensor info is not equal");
+      /* respond deny with src caps string */
+      nns_loge ("Query caps is not acceptable!");
+      nns_loge ("Query client sink caps: %s", gst_caps_to_string (cmd_caps));
```

Comment:
let's try to skip converting caps by using _sdata->src_caps_str and cmd_data.data.data :)